src/lsp.ts

```diff
@@ -0,0 +1,180 @@
+import * as Admzip from "adm-zip";
+import * as which from "which";
+import * as https from "https";
+import * as crypto from "crypto";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import * as vscode from "vscode";
+import { ExtensionContext, WorkspaceConfiguration, workspace } from "vscode";
+import { Executable, LanguageClient, LanguageClientOptions, ServerOptions, TransportKind } from "vscode-languageclient/node";
+
+export abstract class LanguageServerClient {
+  config: WorkspaceConfiguration;
+  ls: LanguageClient | null = null;
+  executableName: string;
+  context: ExtensionContext;
+  private clientOptions: LanguageClientOptions = {
+    documentSelector: ["meson", { "scheme": "file", language: "meson" }]
+  };
+  abstract repoURL: string;
+  abstract get debugExe(): Executable
+  abstract get runExe(): Executable
```

Comment:
Some of these should probably be marked as private/protected. Also ending semicolons

Reply:
Will do that, do you have any linter/formatter for this repo that can do that automatically?

Reply:
@lukester1975 has a PR to add prettier support which I think is a good idea. Then we can also work on adding eslint and various other tools as needed to aid in the development/review process.